### ISSUE TITLE
cachix is broken

### ISSUE THREAD

**Issue by eadwu** [NixOS/nixpkgs#52985]
## Issue description
It might be worth noting that to get to this point, other `haskellPackages` need to patched, https://pastebin.com/raw/1fbxxNie.
Namely its dependency `servant-streaming-client`.
```
src/Servant/Streaming/Client/Internal.hs:50:19: error:
    Variable not in scope:
      runStreamingResponse
        :: a1 -> (GenResponse a0 -> a0) -> IO BS.ByteString
   |
50 |     respStream <- runStreamingResponse <$> streamingRequest req
   |                   ^^^^^^^^^^^^^^^^^^^^

src/Servant/Streaming/Client/Internal.hs:50:44: error:
    Variable not in scope: streamingRequest :: Request -> m a1
   |
50 |     respStream <- runStreamingResponse <$> streamingRequest req
   |                                            ^^^^^^^^^^^^^^^^
```

### Steps to reproduce
```sh
nix-build '<nixpkgs>' -A cachix
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.
```sh
 - system: `"x86_64-linux"`
 - host os: `Linux 4.20.0, NixOS, 19.03.git.c8c39b4 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2pre6526_9f99d624`
 - channels(root): `"nixos-19.03pre162671.44b02b52ea6"`
 - nixpkgs: `$HOME/Downloads/nixpkgs`
```

**Comment by domenkozar** [comment]
See https://github.com/cachix/cachix/pull/149 - I plan to merge that in a week or so.

**Comment by domenkozar** [comment]
For status follow https://github.com/cachix/cachix/pull/163 - blocked by upstream. Best way to install cachix is still via known good set:

```
$ nix-env -iA cachix -f https://cachix.org/api/v1/install
```

**Comment by lheckemann** [comment]
@domenkozar Is installation from nixpkgs a story that you intend to support consistently at some point, or something you're not willing to deal with (understandable)? In the latter case, I'd suggest adding a notice to nixpkgs on the preferred way of installing cachix and closing this issue.

**Comment by domenkozar** [comment]
Yes, I will do my best to support cachix in nixpkgs, but due to Haskell updates work in nixpkgs it's hard to guarantee anything.

I've patched the bug that's affecting the cachix in 673f50f4d95acac75d004e2befccd812b0bd611c and I'll make a release in next couple of days.

**Comment by Moredread** [comment]
@domenkozar will there be a backport to 19.03? Thanks for your work!

**Comment by domenkozar** [comment]
Pushed cachix 0.2.0 to nixpkgs master and 19.03 - it's going to be usable in a couple of hours.
